Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs

```diff
@@ -631,56 +805,16 @@ getNearestDriversCurrentlyOnRide ::
-        &&. personInfo ^. PersonMerchantId ==. val (toKey merchantId)
-        &&. ((Esq.isNothing (driverInfo ^. DriverInformationMode) &&. driverInfo ^. DriverInformationActive) ||. (not_ (Esq.isNothing (driverInfo ^. DriverInformationMode)) &&. (driverInfo ^. DriverInformationMode ==. val (Just DriverInfo.SILENT) ||. driverInfo ^. DriverInformationMode ==. val (Just DriverInfo.ONLINE))))
-        &&. driverInfo ^. DriverInformationOnRide
-        &&. not_ (driverInfo ^. DriverInformationBlocked)
-        &&. ( val (Mb.isNothing mbDriverPositionInfoExpiry)
-                ||. (locationInfo ^. DriverLocationCoordinatesCalculatedAt +. Esq.interval [Esq.SECOND $ maybe 0 getSeconds mbDriverPositionInfoExpiry] >=. val now)
-            )
-        &&. ( Esq.isNothing (val mbVariant) ||. just (vehicleInfo ^. VehicleVariant) ==. val mbVariant -- when mbVariant = Nothing, we use all variants, is it correct?
-                ||. ( case mbVariant of
-                        Just SEDAN ->
-                          driverInfo ^. DriverInformationCanDowngradeToSedan ==. val True
-                            &&. vehicleInfo ^. VehicleVariant ==. val SUV
-                        Just HATCHBACK ->
-                          driverInfo ^. DriverInformationCanDowngradeToHatchback ==. val True
-                            &&. (vehicleInfo ^. VehicleVariant ==. val SUV ||. vehicleInfo ^. VehicleVariant ==. val SEDAN)
-                        Just TAXI ->
-                          driverInfo ^. DriverInformationCanDowngradeToTaxi ==. val True
-                            &&. vehicleInfo ^. VehicleVariant ==. val TAXI_PLUS
-                        _ -> val False
-                    )
-            )
-        &&. (distanceFromDriverToDestination +. distanceFromDestinationToPickup) <. onRideRadius
-    orderBy [asc (distanceFromDestinationToPickup +. distanceFromDriverToDestination)]
-    pure
-      ( personInfo ^. PersonTId,
-        personInfo ^. PersonDeviceToken,
-        personInfo ^. PersonLanguage,
-        driverInfo ^. DriverInformationOnRide,
-        driverInfo ^. DriverInformationCanDowngradeToSedan,
-        driverInfo ^. DriverInformationCanDowngradeToHatchback,
-        driverInfo ^. DriverInformationCanDowngradeToTaxi,
-        locationInfo ^. DriverLocationLat,
-        locationInfo ^. DriverLocationLon,
-        vehicleInfo ^. VehicleVariant,
-        bookingLocationInfo ^. BookingLocationLat,
-        bookingLocationInfo ^. BookingLocationLon,
-        distanceFromDriverToDestination +. distanceFromDestinationToPickup,
-        distanceFromDriverToDestination,
-        driverInfo ^. DriverInformationMode
-      )
+  let onRideRadius = fromIntegral (radiusMeters - reduceRadiusValue) :: Double
+  res <- do
+    drivers <- getDriversWithMerchID merchantId
+    driverQuote <- getDriverQuote drivers
+    driverLocs <- getDriverLocsFromDriverQuote driverQuote mbDriverPositionInfoExpiry LatLong {..} radiusMeters
```

Comment:
You need to lookup drivers nearby first, then rest.
Otherwise the above two queries will list the full table